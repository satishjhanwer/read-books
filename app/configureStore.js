/**
 * Create the store with dynamic reducers
 */
import { applyMiddleware, createStore } from 'redux';
import { composeWithDevToolsDevelopmentOnly } from '@redux-devtools/extension';
import createSagaMiddleware from 'redux-saga';
import { createReduxHistoryContext } from 'redux-first-history';
import { persistStore } from 'reduxjs-toolkit-persist';
import { createBrowserHistory } from 'history';

import createReducer from './reducers';

const initialState = {};

const { createReduxHistory, routerMiddleware, routerReducer } = createReduxHistoryContext({
  history: createBrowserHistory(),
});

const reduxSagaMonitorOptions = {};
const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);

export const store = createStore(
  createReducer(routerReducer),
  initialState,
  composeWithDevToolsDevelopmentOnly(applyMiddleware(...[sagaMiddleware, routerMiddleware])),
);

export const persistor = persistStore(store);
export const history = createReduxHistory(store);
