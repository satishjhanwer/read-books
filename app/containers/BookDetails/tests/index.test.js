/**
 * Test the BookDetails
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { BookDetails } from '../index';
import configureStore from '../../../configureStore';

describe('<BookDetails />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <BookDetails match={{ params: {} }} />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
