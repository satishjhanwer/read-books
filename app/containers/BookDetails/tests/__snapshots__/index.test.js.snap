// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookDetails /> should render and match the snapshot 1`] = `null`;
